opentelemetry-api/src/opentelemetry/propagators/composite.py

```diff
@@ -65,3 +65,18 @@ def inject(
         """
         for propagator in self._propagators:
             propagator.inject(set_in_carrier, carrier, context)
+
+    @property
+    def fields(self) -> typing.Set[str]:
+        """Returns a set with the fields set in `inject`.
+
+        See
+        `opentelemetry.trace.propagation.textmap.TextMapPropagator.fields`
+        """
+        composite_fields = set()
+
+        for propagator in self._propagators:
+            for field in propagator.fields:
+                composite_fields.add(field)
```

Comment:
is the behaviour for the composite propagator specified in the spec?

Reply:
I don't think it is, I tried to replicate the Go implementation with one particular difference. In this implementation a `set` is returned instead of a `list`. There is no defined order for these fields (these fields may be called in `inject` in an arbitrary order and even if this order is constant it has no relation with the fields themselves), so an ordered sequence like a list makes no sense. Also, (if I understand correctly) the [intention](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/context/api-propagators.md#fields) of this attribute is to allow the user to tell if a certain field is also used in `inject`. That is an `in` operation that has a better (O(1)) performance in a `set` than in a list (O(N)).